compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -462,6 +462,42 @@ pub struct CReaderCacheKey {
 #[rustc_pass_by_value]
 pub struct Ty<'tcx>(Interned<'tcx, WithCachedTypeInfo<TyKind<'tcx>>>);
 
+pub fn maybe_not_supported_generic_const_param(ty: Ty<'_>) -> Option<bool> {
```

Comment:
questions:
1. I've chosen to use ﻿"maybe not supported" because we require `﻿CompileTimeEvalContext` to obtain exact information. I have represented potential lack of support using `﻿Option<bool>`. In this case, `﻿Some(true)` indicates a lack of support and `﻿None` stands for an unknown state. Is this approach appropriate?
2. The placement of ﻿`pub fn maybe_not_supported_generic_const_param() { }` in ﻿`rustc_middle::ty` seems a bit odd. Should we consider moving it to another location?